ArgumentParser is not subscriptable error when an invalid argument is provided to fprime-util impl

| | |
|:---|:---|
|**_F´ Version_**| v3.5.1 |
|**_Affected Component_**| fprime-util |
---
## Problem Description

When you pass an invalid option to `fprime-util impl`, it is expected that it reports an error. However, in addition to reporting the error, it also unexpectedly displays two exceptions:

```
# fprime-util impl --overwrite
[ERROR] 'fprime-util impl' supplied invalid arguments: --overwrite

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/fprime/util/cli.py", line 364, in parse_args
    cmake_args, make_args = validate(parsed, unknown)
  File "/usr/local/lib/python3.9/site-packages/fprime/util/cli.py", line 280, in validate
    raise ArgValidationException(msg)
fprime.util.cli.ArgValidationException: 'fprime-util impl' supplied invalid arguments: --overwrite

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/fprime-util", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/fprime/util/__main__.py", line 15, in main
    return fprime.util.cli.utility_entry(args=sys.argv[1:])
  File "/usr/local/lib/python3.9/site-packages/fprime/util/cli.py", line 34, in utility_entry
    parsed, cmake_args, make_args, parser, runners = parse_args(args)
  File "/usr/local/lib/python3.9/site-packages/fprime/util/cli.py", line 367, in parse_args
    parsers.get(parsed.command, (parser,))[0].print_usage()
TypeError: 'ArgumentParser' object is not subscriptable
```

## Context / Environment

Execute `fprime-util version-check` and share the output.

```
# fprime-util version-check
Operating System: Linux
CPU Architecture: x86_64
Platform: Linux-4.18.0-553.34.1.el8_10.x86_64-x86_64-with-glibc2.34
Python version: 3.9.21
CMake version: 3.26.5
Pip version: 21.3.1
Pip packages:
    fprime-tools==3.5.1
    fprime-gds==3.5.2a1
    fprime-fpp-*==2.3.0a1
Project submodules:
    https://github.com/nasa/fprime.git @ v3.5.1-26-gc68c32a9c
```

## How to Reproduce

1. Run `fprime-util impl --overwrite`

## Expected Behavior

This command should only print the following output:

```
# fprime-util impl --overwrite
[ERROR] 'fprime-util impl' supplied invalid arguments: --overwrite
```

--- a/src/fprime/fpp/visualize.py
+++ b/src/fprime/fpp/visualize.py
@@ -92,7 +92,7 @@ def run_fprime_visualize(
                 f"Unable to write to {viz_cache_base.resolve()}. Use --working-dir to set a different location."
             )
         for layout_txt in layout_txt_file_match:
-            print(f"Generated layout TXT file: {layout_txt.resolve()}")  
+            print(f"Generated layout TXT file: {layout_txt.resolve()}")
             connection_graph_json = viz_cache / f"{layout_txt.stem}.json"
             # Execute: fpl-layout < ConnectionGraph.txt > ConnectionGraph.json
             with open(connection_graph_json.resolve(), "w") as json_file:
@@ -100,12 +100,18 @@ def run_fprime_visualize(
                     txt_contents = txt_file.read()
                     topology_connections += txt_contents
                     subprocess.run(
-                        ["fpl-layout"], stdout=json_file, input=txt_contents.encode(), check=True
+                        ["fpl-layout"],
+                        stdout=json_file,
+                        input=txt_contents.encode(),
+                        check=True,
                     )
         # Generate layout JSON for entire topology (all connections graphs in one layout)
         with open(topology_json.resolve(), "w") as json_file:
             subprocess.run(
-                ["fpl-layout"], stdout=json_file, input=topology_connections.encode(), check=True
+                ["fpl-layout"],
+                stdout=json_file,
+                input=topology_connections.encode(),
+                check=True,
             )
         source_dirs.append(viz_cache)
     source_resolved = [str(source.resolve()) for source in source_dirs]

--- a/src/fprime/util/cli.py
+++ b/src/fprime/util/cli.py
@@ -364,7 +364,11 @@ def parse_args(args):
         cmake_args, make_args = validate(parsed, unknown)
     except ArgValidationException as exc:
         print(f"[ERROR] {exc}", end="\n\n")
-        parsers.get(parsed.command, (parser,))[0].print_usage()
+        selected_parser = parsers.get(parsed.command, parser)
+        if type(selected_parser) is tuple:
+            selected_parser[0].print_usage()
+        else:
+            selected_parser.print_usage()
         sys.exit(1)
     return parsed, cmake_args, make_args, parser, runners
 

